hooks/post_gen_project.py

```diff
@@ -205,7 +205,7 @@ def update_local_workflows(central_workflows):
 def main():
     """Execute when user runs cookiecutter."""
     if "." in "{{ cookiecutter.project_name }}":
-        add_supermodules(ROOT, "{{ cookiecutter.project_name }}")
+        add_supermodules(ROOT, "{{ cookiecutter.package_dir_name }}")
```

Comment:
Then, we need to use `package_dir_name` which is ex) `src/diffpy/my_project` rather than `project_name` which becomes `src/diffpy/my-project`. 

I have tested locally with this new change.